### PR TITLE
Move PEP command and utils to own cog, fix bug related not checking status and add GitHub token

### LINKED ISSUE #1335: refresh_peps_urls should check the response status code
Sentry Issue: [BOT-JD](https://sentry.io/organizations/python-discord/issues/2103781334/?referrer=github_integration)

In the case below, the GitHub rate limit was exceed and a 403 was returned.

```
TypeError: string indices must be integers
  File "bot/exts/utils/utils.py", line 224, in refresh_peps_urls
    name = file["name"]

Task exception was never retrieved
future: <Task finished name='Task-10' coro=<Utils.refresh_peps_urls() done, defined at /bot/bot/exts/utils/utils.py:212> exception=TypeError('string indices must be integers')>
```

### LINKED ISSUE #1336: validate_pep_number should acount for last_refreshed_peps being None
Sentry Issue: [BOT-JE](https://sentry.io/organizations/python-discord/issues/2103784890/?referrer=github_integration)

last_refreshed_peps  may be unset if an error was encountered with `refresh_peps_urls`, such as the rate limit in #1335

```
TypeError: unsupported operand type(s) for +: 'NoneType' and 'datetime.timedelta'
  File "discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "bot/exts/utils/utils.py", line 244, in pep_command
    if not (pep_embed := await self.validate_pep_number(pep_number)):
  File "bot/exts/utils/utils.py", line 272, in validate_pep_number
    and (self.last_refreshed_peps + timedelta(minutes=30)) <= datetime.now()

Error executing command invoked by <REDACTED>: !pep 8
```

### PR DESCRIPTION
**Pull request by ks129**
**Note:** Before merging this PR, DevOps team should add `GITHUB_API_TOKEN` environment variable.

Should be merged fast, PEP command is broken currently because of rate limits.

- Moved PEP related utilities and command to own cog.
- Make last refresh non-optional.
- Catch non-200 responses.
- Add GitHub token.
- Closes #1335
- Closes #1336

### CODE REVIEW

File: bot/exts/utils/pep.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+GITHUB_API_HEADERS = {}
+if Keys.github:
+    GITHUB_API_HEADERS["Authorization"] = f"token {Keys.github}"
+
+
+class PythonEnhancementProposals(Cog):
+    """Cog for displaying information about PEPs."""
+
+    BASE_PEP_URL = "http://www.python.org/dev/peps/pep-"
+    PEPS_LISTING_API_URL = "https://api.github.com/repos/python/peps/contents?ref=master"
```
Comment (Xithrius) [inline bot/exts/utils/pep.py:29]: I feel like these constants should be outside the class, next to `ICON_URL`.

File: bot/exts/utils/pep.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+    async def refresh_peps_urls(self) -> None:
+        """Refresh PEP URLs listing in every 3 hours."""
+        # Wait until HTTP client is available
+        await self.bot.wait_until_ready()
+        log.trace("Started refreshing PEP URLs.")
+        self.last_refreshed_peps = datetime.now()
+
+        async with self.bot.http_session.get(
+            self.PEPS_LISTING_API_URL,
```
Comment (Xithrius) [inline bot/exts/utils/pep.py:46]: If the constants are moved outside the class, this should be renamed.
```suggestion
           PEPS_LISTING_API_URL,
```

File: bot/exts/utils/pep.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            )
+
+        return None
+
+    def generate_pep_embed(self, pep_header: Dict, pep_nr: int) -> Embed:
+        """Generate PEP embed based on PEP headers data."""
+        # Assemble the embed
+        pep_embed = Embed(
+            title=f"**PEP {pep_nr} - {pep_header['Title']}**",
+            description=f"[Link]({self.BASE_PEP_URL}{pep_nr:04})",
```
Comment (Xithrius) [inline bot/exts/utils/pep.py:103]: Same here
```suggestion
            description=f"[Link]({BASE_PEP_URL}{pep_nr:04})",
```

**Review by Xithrius — APPROVED** [review]
LGTM!

**Review by Den4200 — CHANGES_REQUESTED** [review]
This cog would better fit into the `info` directory.

**Comment by ks129** [comment]
Moved PEP cog to info extensions group.

**Review by Den4200 — APPROVED** [review]
Thanks!

**Pull request merged**